File: fnc/examples/chapter10/shootmems.m
%%
% We eliminate the guesswork for $w(0)$ and let |shoot| do the work for us.
lambda = 0.6;
phi = @(r,w,dwdr) lambda./w.^2 - dwdr./r;   
a = eps;  b = 1;    % avoid r=0 in denominator

%%
% We specify the given and unknown endpoint values.
lval = [];  lder = 0;   % w(a)=?, w'(a)=0
rval = 1;   rder = [];  % w(b)=1, w'(b)=?

%%
[r,w,dwdx] = shoot(phi,[a,b],lval,lder,rval,rder,0.8);
plot(r,w)
title('Correct solution')   % ignore this line
xlabel('x'), ylabel('w(x)')   % ignore this line

%%
% The value of $w$ at $r=1$, meant to be exactly one, was computed to be
format long
w(end)

%%
% The accuracy is consistent with the error tolerance used for |ode45| by
% |shoot|. The initial value $w(0)$ that gave this solution is
w(1)

File: julia/scripts/chapter10/shootmems.jl

lambda = 0.6
phi = (r,w,dwdr) -> lambda/w^2 - dwdr/r;
a = eps();  b = 1;

lval = [];  lder = 0;   # w(a)=?, w'(a)=0
rval = 1;   rder = [];  # w(b)=1, w'(b)=?

include("../FNC.jl")
r,w,dwdx = FNC.shoot(phi,(a,b),lval,lder,rval,rder,0.8)

using Plots
plot(r,w,leg=:none,title="Correct solution",
    xaxis=("x"), yaxis=("w(x)"))

@show w[end];

@show w[1];
